extensions/strategies/trust_distrust/strategy.js

```diff
@@ -0,0 +1,136 @@
+var z = require('zero-fill')
+  , n = require('numbro')
+
+module.exports = function container (get, set, clear) {
+  return {
+    name: 'trust_distrust',
+    description: 'Sell when price higher than $sell_min% and highest point - $sell_threshold% is reached. Buy when lowest price point + $buy_threshold% reached.',
+
+    getOptions: function () {
+      this.option('period', 'period length', String, '30m')
+      this.option('min_periods', 'min. number of history periods', Number, 52)
+      this.option('sell_threshold', 'sell when the top drops at least below this percentage', Number, 2)
+      this.option('sell_threshold_max', 'sell when the top drops lower than this max, regardless of sell_min (panic sell, 0 to disable)', Number, 0)
+      this.option('sell_min', 'do not act on anything unless the price is this percentage above the original price', Number, 1)
+      this.option('buy_threshold', 'buy when the bottom increased at least above this percentage', Number, 2)
+      this.option('buy_threshold_max', 'wait for multiple buy signals before buying (kill whipsaw, 0 to disable)', Number, 0)
+      this.option('greed', 'sell if we reach this much profit (0 to be greedy and either win or lose)', Number, 0)
+    },
+
+    calculate: function (s) {
+      if (typeof s.trust_distrust_start_greed === 'undefined') {
+        s.trust_distrust_start_greed = s.period.high
+      }
+      if (typeof s.trust_distrust_start === 'undefined') {
+        s.trust_distrust_start = s.period.high
+      }
+      if (typeof s.trust_distrust_highest === 'undefined') {
+        s.trust_distrust_highest = s.period.high
+      }
+      if (typeof s.trust_distrust_lowest === 'undefined') {
+        s.trust_distrust_lowest = s.period.high
+      }
+      if (typeof s.trust_distrust_last_action === 'undefined') {
+        s.trust_distrust_last_action = null
+      }
+      if (typeof s.trust_distrust_buy_threshold_max === 'undefined') {
+        s.trust_distrust_buy_threshold_max = 0
+      }
+
+      // when our current price is higher than what we recorded, overwrite
+      if (s.period.high > s.trust_distrust_highest) {
+        s.trust_distrust_highest = s.period.high
+      }
+
+      // when our current price is lower than what we recorded, overwrite
+      if (s.trust_distrust_lowest > s.period.high) {
+        s.trust_distrust_lowest = s.period.high
+      }
+    },
```

Comment:
note to self: might want to consider calculating EMA/MACD (both?) to check if we are in an up or down trend, and advise buy/sell accordingly.

Reply:
Hi @DeviaVir, did you manage to implement this idea of yours?